src/index.js

```diff
@@ -0,0 +1,324 @@
+            });
+
+        })
+    }
+    async get_id() {
+        return Math.round(Math.random() * 100000000 + 1);
+    }
+    async fetch_ids() {
+        
+        let app = await BeetClientDB.apps.where("apphash").equals(this.identity.apphash).first();        
+        let id = app.next_id;
+        let new_id = await this.next_id();
+        let next_hash = await CryptoJS.SHA256('' + new_id).toString();
+        return {
+            id: id,
+            next_hash: next_hash.toString()
+        };
+    }
+    async next_id() {
+        if (this.connected && this.authenticated && this.linked) {
+            let next_id = Math.round(Math.random() * 100000000 + 1);
+            await BeetClientDB.apps.where("apphash").equals(this.identity.apphash).modify({
+                next_id: next_id
+            });
+            return next_id;
+        } else {
+            throw new Error("You must be connected, authorised and linked.");
+        }
+    }
+    async link(chain = 'BTS') {
+        return new Promise(async (resolve, reject) => {
+            if (!this.connected) throw new Error("You must connect to Beet first.");
+            if (!this.initialised) throw new Error("You must initialise the Beet Client first via init(appName).");            
+            setTimeout(() => {
+                resolve(false);
+            }, this.options.linkTimeout);
+            this.privk = crypto.randomBytes(32);
+            let pubkey = await eccrypto.getPublic(this.privk).toString('hex');
+            this.secret = await eccrypto.derive(this.privk, Buffer.from(this.beetkey, 'hex'));
+            var next_id = Math.round(Math.random() * 100000000 + 1);
+            this.chain = "BTS";
+            var next_hash = await CryptoJS.SHA256('' + next_id);
+            let linkobj = {
+                chain: "BTS",
+                pubkey: pubkey,
+                next_hash: next_hash.toString()
+            }
+            var link = this.sendRequest('link', linkobj);
+            link.then(async res => {
+                this.apphash = CryptoJS.SHA256(this.detected.name + ' ' + this.origin + ' ' + this.appName + ' ' + this.chain + ' ' + res.account_id).toString();
+                this.appstore = await BeetClientDB.apps.add({
+                    keyhash: this.keyhash,
+                    apphash: this.apphash,
+                    account_id: res.account_id,
+                    chain: this.chain,
+                    appName: this.appName,
+                    secret: this.secret.toString('hex'),
+                    next_id: next_id
+                });
+                this.authenticated = res.authenticate;
+                this.linked = res.link;
+                this.identity = await BeetClientDB.apps.where("apphash").equals(this.apphash).first();
+
+                this.otp = new OTPAuth.HOTP({
+                    issuer: "Beet",
+                    label: "BeetAuth",
+                    algorithm: "SHA1",
```

Comment:
Why not SHA256?

Reply:
It was the default and there is no real reason to change it as it's a one-time hash and not the primary authentication mechanism (see: https://www.quora.com/Why-is-the-SHA1-algorithm-still-being-used-with-2FA-codes-instead-of-SHA2)